### PR TITLE
Niganesh/move jwt validation to network layer

### PR DESCRIPTION
**Pull request by nithyaganeshng**
Moved signature verification and parsing the jws token to network layer. After making a network call, if the response is successful, signature is verified, token received is parsed and corresponding object is returned. 

Validation:
Verified unit and instrumented tests passed
Manually tested issuance and presentation flows

**Type of change:**
- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt [hunk 1]
```diff
@@ -28,21 +29,17 @@ class PresentationService @Inject constructor(
...
     suspend fun getRequest(stringUri: String): Result<PresentationRequest> {
         return runResultTry {
             val uri = verifyUri(stringUri)
-            val requestToken = getPresentationRequestToken(uri).abortOnError()
-            val tokenContents =
-                serializer.parse(
-                    PresentationRequestContent.serializer(),
-                    JwsToken.deserialize(requestToken, serializer).content()
-                )
+            val tokenContents = getPresentationRequestToken(uri).abortOnError()
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt:40]: nit, consolidate naming between `tokenContents` and `getPresentationRequestToken` (consider new return type)

File: sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt [hunk 2]
```diff
@@ -56,15 +53,13 @@ class PresentationService @Inject constructor(
...
     }
 
-    private suspend fun getPresentationRequestToken(uri: Uri): Result<String> {
+    private suspend fun getPresentationRequestToken(uri: Uri): Result<PresentationRequestContent> {
         val serializedToken = uri.getQueryParameter("request")
-        if (serializedToken != null) {
-            return Result.Success(serializedToken)
-        }
+        if (serializedToken != null)
+            return Result.Success(unwrapPresentationRequest(serializedToken))
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt:59]: in this scenario the signature is not verified (but it is in the request_uri scenario within NetworkOp.

Maybe also consolidate this code. Though I'm not seeing a super straight forward way to do so. Maybe you have a good idea.

File: sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt [hunk 3]
```diff
@@ -5,11 +5,35 @@
...
+import com.microsoft.did.sdk.crypto.protocols.jose.jws.JwsToken
 import com.microsoft.did.sdk.datasource.network.GetNetworkOperation
 import com.microsoft.did.sdk.datasource.network.apis.ApiProvider
+import com.microsoft.did.sdk.util.controlflow.InvalidSignatureException
+import com.microsoft.did.sdk.util.controlflow.PresentationException
+import com.microsoft.did.sdk.util.controlflow.Result
+import com.microsoft.did.sdk.util.serializer.Serializer
 import retrofit2.Response
 
 //TODO("override onSuccess method to create receipt when this is spec'd out")
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt:19]: nit, I think this comment is outdated?

**Review by AndiMiko — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt (hunk 3 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt]: yeah it was semi outdated :) I have updated the comment now.

File: sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt (hunk 2 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt]: Oh this scenario is when the request is retrieved directly from query parameter without having to make a network call. Added the same method here. Followed the rule that 2 times is not a pattern and duplicated the code here.

File: sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt (hunk 1 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt]: I renamed the variable tokenContents to align with return type. Is that what you meant?

File: sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt (hunk 3 above)
Comment (AndiMiko) [reply sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt]: you changed the comment, but I thought we will never have receipts present in our SDK or am I missunderstanding what this comment is saying?

File: sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt (hunk 1 above)
Comment (AndiMiko) [reply sdk/src/main/java/com/microsoft/did/sdk/PresentationService.kt]: yes, but I meant that the method name also probably should be called `getPresentationRequestContent` then :) it was the token before

**Review by AndiMiko — APPROVED** [review]

File: sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt (hunk 3 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt]: This is the receipts we would receive from VC SDK on presenting a VC. We would process those signed receipts here and return corresponding objects to layer above for further processing.

**Pull request merged**